src/ucp/rma/flush.c

```diff
@@ -444,17 +446,25 @@ static unsigned ucp_worker_flush_progress(void *arg)
     ucs_status_t status;
     ucp_ep_h ep;
 
-    status = ucp_worker_flush_check(worker);
-    if ((status == UCS_OK) || (&next_ep->ep_list == &worker->all_eps)) {
-        /* If all ifaces are flushed, or we finished going over all endpoints,
-         * no need to progress this request actively any more. Just wait until
-         * all associated endpoint flush requests are completed.
-         */
-        ucp_worker_flush_complete_one(req, UCS_OK, 1);
-    } else if (status != UCS_INPROGRESS) {
-        /* Error returned from uct iface flush */
-        ucp_worker_flush_complete_one(req, status, 1);
-    } else if (worker->context->config.ext.flush_worker_eps) {
+    if (ucp_worker_flush_ops_count_check(worker)) {
```

Comment:
why not leaving this check inside `ucp_worker_flush_check`? seems unnecessary
also checking  `!worker->flush_ops_count` looks more clear than `ucp_worker_flush_ops_count_check` name

Reply:
fixed